### PR TITLE
feat: mantine viz menus with performance improvements and bugfixes

### PR DESCRIPTION
**Pull request by IrakliJani**
Closes: https://github.com/lightdash/lightdash/issues/8263
Closes: https://github.com/lightdash/lightdash/issues/6339

### Description:

- reverts back revert of viz context menus.
- adds bug fix for nested menus (it's flat now, see screenshot).
- adds performance improvement by separating menu component and only rendering it once if cell is selected.

CleanShot 2023-12-12 at 00 00 33@2x

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/components/common/Table/BodyCell.tsx [hunk 1]
```diff
@@ -23,97 +24,72 @@ interface CommonBodyCellProps {
...
-                    )}
-                />
-            )}
-        />
+        <Tooltip
+            withinPortal
+            position="top"
+            disabled={!tooltipContent || minimal}
+            label={tooltipContent}
+        >
```
Comment (joaoviana) [inline packages/frontend/src/components/common/Table/BodyCell.tsx:71]: could we also do something about the rendering of these tooltips? 
For example, similar to the `Menu` position calculation or maybe having one `Tooltip` shared across the cells? 🤔

**Review by joaoviana — COMMENTED** [review]
Code looks great, just one observation on Tooltips!

**Comment by IrakliJani** [comment]
CleanShot 2023-12-12 at 16 09 19@2x

https://github.com/lightdash/lightdash/assets/962095/636f0ac6-a7b5-4a46-891f-141fea6d40aa

https://github.com/lightdash/lightdash/assets/962095/44691db4-82c7-4127-8aca-d25a1727ecd9


react render tree:

CleanShot 2023-12-12 at 16 13 56@2x

**Comment by IrakliJani** [comment]
I've moved menus to the same level as tooltips. so now when you open a menu it won't rerender whole table.


https://github.com/lightdash/lightdash/assets/962095/e4d6ba47-551e-45db-a920-7f7c54c1b477

**Comment by IrakliJani** [comment]
before (blueprint)

image


after + optimizations (mantine)
image

**Comment by IrakliJani** [comment]
testing on render:

both tables have the same dimensions + metrics selected and same conditional filtering applied.

blueprint table: https://lightdash-pr-8256.onrender.com/projects/3675b69e-8324-4110-bdca-059031aa8da3/saved/1b8833ca-1a29-48c8-b884-801e75fe5820/view

mantine table (optimized): https://lightdash-pr-8271.onrender.com/projects/3675b69e-8324-4110-bdca-059031aa8da3/saved/cbd22dae-72f7-4870-8d75-dcd0933971e6/view


test 1:

- load links above.
- collapse all cards.
- start performance recording
- open results card
- open charts card
- stop recording
- repeat the same step 2 times for each link

results:

blueprint:

CleanShot 2023-12-13 at 13 30 23@2x
CleanShot 2023-12-13 at 13 32 11@2x

mantine:

CleanShot 2023-12-13 at 13 29 02@2x
CleanShot 2023-12-13 at 13 36 15@2x

**Review by joaoviana — COMMENTED** [review]
Some differences on Cell colours between this Render env and another without these changes
Why are these blue here? was there some tweaking added to the table when testing it?

image (3)
image (2)

**Comment by IrakliJani** [comment]
# test 2

- load links from previous comment.
- chart result is open
- start recording
- start scrolling till the end (do not hover on cells with tooltips while scrolling)
- stop recording

| mantine | blueprint |
| ------- | --------- |
|  |  |
|  |  |
|  |  |
|  |  |
|  |  |
|  |  |

**Review by joaoviana — APPROVED** [review]

**Comment by IrakliJani** [comment]
# test 3
## hover test



| mantine 🏅 winner  | blueprint |
| ------- | --------- |
| CleanShot 2023-12-13 at 14 46 13@2x | CleanShot 2023-12-13 at 14 46 22@2x |
| CleanShot 2023-12-13 at 14 48 24@2x | CleanShot 2023-12-13 at 14 48 20@2x |
| CleanShot 2023-12-13 at 15 05 12@2x | CleanShot 2023-12-13 at 15 05 06@2x |

**Comment by IrakliJani** [comment]
# test 4
## open menu test

- load links from previous comment.
- chart result is open
- start recording
- click dimension without tooltip (opens menu)
- click dimension with tooltip (opens menu)
- click metric without tooltip (opens menu (with more options))
- stop recording



| mantine 🏋️  winner | blueprint |
| ------- | --------- |
| CleanShot 2023-12-13 at 15 10 42@2x | CleanShot 2023-12-13 at 15 10 46@2x |
| CleanShot 2023-12-13 at 15 11 53@2x | CleanShot 2023-12-13 at 15 11 59@2x |
| CleanShot 2023-12-13 at 15 16 17@2x | CleanShot 2023-12-13 at 15 16 21@2x |

**Comment by IrakliJani** [comment]
# test 5

## dashboard load test

exactly same dashboards with exactly same tables:

mantine: https://lightdash-pr-8271.onrender.com/projects/3675b69e-8324-4110-bdca-059031aa8da3/dashboards/61f76084-bcd2-41d1-bad9-74b31a90c8ff/view

blueprint: https://lightdash-pr-8256.onrender.com/projects/3675b69e-8324-4110-bdca-059031aa8da3/dashboards/2c9378db-3bd5-4dda-a201-bbde9971d800/view

steps:
- reload with performance monitor

| mantine 🍾 winner | blueprint |
| ------- | --------- |
| CleanShot 2023-12-13 at 15 24 51@2x | CleanShot 2023-12-13 at 15 24 57@2x |

**Pull request merged**